test/reservation_desk_test.rb

```diff
@@ -0,0 +1,317 @@
+require_relative "test_helper"
+
+describe "ReservationDesk class" do
+  before do
+    @reservation_desk = Hotel::ReservationDesk.new()
+  end
+
+  describe "ReservationDesk instantiation" do
+    it "is an instance of ReservationDesk" do
+      expect(@reservation_desk).must_be_kind_of Hotel::ReservationDesk
+    end
+  end
+
+  describe "rooms" do
+    it "returns an array" do
+      expect(@reservation_desk.rooms).must_be_kind_of Array
+    end
+
+    it "returns an array of Rooms" do
+      expect(@reservation_desk.rooms[0]).must_be_kind_of Hotel::Room
+      expect(@reservation_desk.rooms.last).must_be_kind_of Hotel::Room
+    end
+
+    it "includes 20 rooms by default" do
+      expect(@reservation_desk.rooms.length).must_equal 20
+    end
+
+    it "the first room has an id of 1" do
+      expect(@reservation_desk.rooms[0].id).must_equal 1
+    end
+
+    it "the last room has an id which is equal to the number of rooms" do
+      reservation_desk = Hotel::ReservationDesk.new(room_num: 134)
+      expect(reservation_desk.rooms.last.id).must_equal 134
+    end
+  end
+
+  describe "find_room_by_id" do
+    it "returns an instanse of Room" do
+      expect(@reservation_desk.find_room_by_id(2)).must_be_kind_of Hotel::Room
+    end
+
+    it "returns a Room with a matching ID" do
+      expect(@reservation_desk.find_room_by_id(2).id).must_equal 2
+    end
+
+    it "returns nill if there is no Room with a matching ID" do
+      assert_nil @reservation_desk.find_room_by_id(20000000)
+   end
+  end
+
+  describe "find_reservations" do
```

Comment:
These tests are quite thorough. Nice work! Make sure to also include a test for an edge case such as no reservations for a given date range.